File: backend/LPCylinderMES.Api/Models/Item.cs
using System;
using System.Collections.Generic;

namespace LPCylinderMES.Api.Models;

public partial class Item
{
    public int Id { get; set; }

    public string ItemNo { get; set; } = null!;

    public string? ItemDescription { get; set; }

    public string ItemType { get; set; } = null!;

    public int RequiresSerialNumbers { get; set; }

    public string? ProductLine { get; set; }

    public bool? RequiresGaugeOption { get; set; }

    public bool? RequiresFillerOption { get; set; }

    public bool? RequiresCollarOption { get; set; }

    public bool? RequiresFootRingOption { get; set; }

    public bool? RequiresValveTypeOption { get; set; }

    public string? SystemCode { get; set; }

    public int? ItemSize { get; set; }

    public virtual ICollection<CustomerItem> CustomerItems { get; set; } = new List<CustomerItem>();

    public virtual ItemSize? ItemSizeNavigation { get; set; }

    public virtual ICollection<Pricing> Pricings { get; set; } = new List<Pricing>();

    public virtual ICollection<SalesOrderDetailCharge> SalesOrderDetailCharges { get; set; } = new List<SalesOrderDetailCharge>();

    public virtual ICollection<SalesOrderDetail> SalesOrderDetails { get; set; } = new List<SalesOrderDetail>();
}

File: frontend/src/types/item.ts
export interface ItemListItem {
  id: number;
  itemNo: string;
  itemDescription: string | null;
  itemType: string;
  productLine: string | null;
  sizeName: string | null;
  basePrice: number | null;
}

export interface ItemDetail {
  id: number;
  itemNo: string;
  itemDescription: string | null;
  itemType: string;
  productLine: string | null;
  itemSizeId: number | null;
  sizeName: string | null;
  systemCode: string | null;
  requiresSerialNumbers: number;
  requiresGaugeOption: boolean | null;
  requiresFillerOption: boolean | null;
  requiresCollarOption: boolean | null;
  requiresFootRingOption: boolean | null;
  requiresValveTypeOption: boolean | null;
  pricings: PricingRecord[];
  crossReferences: CrossReference[];
}

export interface ItemCreate {
  itemNo: string;
  itemDescription?: string | null;
  itemType: string;
  productLine?: string | null;
  itemSizeId?: number | null;
}

export interface ItemUpdate {
  itemNo: string;
  itemDescription: string | null;
  itemType: string;
  productLine: string | null;
  itemSizeId: number | null;
  systemCode: string | null;
  requiresSerialNumbers: number;
  requiresGaugeOption: boolean | null;
  requiresFillerOption: boolean | null;
  requiresCollarOption: boolean | null;
  requiresFootRingOption: boolean | null;
  requiresValveTypeOption: boolean | null;
}

export interface PricingRecord {
  id: number;
  effectiveDate: string;
  unitPrice: number | null;
  notes: string | null;
  itemId: number;
  customerId: number | null;
  customerName: string | null;
}

export interface PricingCreate {
  effectiveDate: string;
  unitPrice: number | null;
  notes?: string | null;
  customerId?: number | null;
}

export interface CrossReference {
  id: number;
  lpcItemNumber: string;
  erpItemNumber: string;
}

export interface CrossReferenceCreate {
  lpcItemNumber: string;
  erpItemNumber: string;
}

export interface ItemSizeLookup {
  id: number;
  name: string;
  size: number;
}
